src/Core/src/Hosting/AppHostBuilder.cs

```diff
@@ -52,15 +52,18 @@ public IAppHost Build()
 			CreateHostBuilderContext();
 			BuildAppConfiguration();
 
+			_services = _serviceCollectionFactory();
 			if (_services == null)
 				throw new InvalidOperationException("The ServiceCollection cannot be null");
 
 			BuildServiceCollections(_services);
+			BuildServices(_services);
 
-			CreateServiceProvider(_services);
+			_services.TryAddSingleton<ILoggerFactory, FallbackLoggerFactory>();
```

Comment:
The `FallbackLoggerFactory` is only added if no other loggers are found. This way a default implementation has some debug logging.